zerver/models/users.py

```diff
@@ -1211,3 +1211,27 @@ def get_bot_dicts_in_realm(realm: "Realm") -> list[dict[str, Any]]:
 
 def is_cross_realm_bot_email(email: str) -> bool:
     return email.lower() in settings.CROSS_REALM_BOT_EMAILS
+
+
+class ExternalAuthID(models.Model):
+    user_profile = models.ForeignKey(UserProfile, on_delete=CASCADE)
```

Comment:
We can just call this `user`; we'll rename `UserProfile` eventually, and might as well not add to the headache.